heat/engine/service.py

```diff
@@ -812,7 +812,15 @@ def _stack_create(stack, msg_queue=None):
             stack_user_project_id, convergence, parent_resource_name,
             template_id)
 
+        # Return without allocating resources if stack set to initialize
+        if stack.initialize:
```

Comment:
Looking through this code, I'm wondering if we need to define a new action at all? What if it was just:

```suggestion
         if stack.initialize:
             stack.store()
             return dict(stack.identifier())
```

I would expect it to have action = CREATE, status = PENDING or whatever the initial status is, which would make good semantic sense for our purposes. That seems like it could be cleaner, also we understand how Horizon would handle displaying that -- I'm not sure how Horizon will deal with a stack in a new state, perhaps we'd need to add support for it.
         
```

Reply:
OK, actually it looks like we need to move this around a bit. This aborts before the Heat user is created, which is used to hold the permissions etc to launch the stack.
Probably we will have to do something like this around the final if/else block here.

```python
if not stack.initialize:
    if convergence:
        // ...
    else:
        // ...
```